In [1]:
import numpy as np
from scipy.special import factorial

def combinations(n, k):
    """сочетания"""
    return int(factorial(n) / (factorial(n - k) * factorial(k)))

### 1) Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) а) среднее арифметическое, б) среднее квадратичное отклонение, в) смещенную и несмещенную оценки дисперсий для данной выборки.

In [2]:
a = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
a

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

**а) Среднее арифметическое $$\mu=\frac{1}{n} \sum_{i=1}^{n}x_{i}$$**

In [3]:
n = len(a)
mean = a.sum()/n
mean

65.3

**б) Cреднее квадратичное отклонение $$S=\sqrt{\frac{\sum_{i=1}^{n}(x_{i}-\bar{x})^{2}}{n}}$$**

In [4]:
std = np.sqrt(np.sum((a - mean)**2)/n)
std

30.823854398825596

**в) смещенная оценка дисперсии $$S^{2}=\frac{\sum_{i=1}^{n}(x_{i}-\bar{x})^{2}}{n}$$**

In [5]:
variance1 = np.sum((a - mean)**2)/n
variance1

950.11

**г) несмещенная оценка дисперсии $$S_{unbiased}^{2}=\frac{\sum_{i=1}^{n}(x_{i}-\bar{x})^{2}}{n-1}$$**

In [6]:
variance2 = np.sum((a - mean)**2)/(n-1)
variance2

1000.1157894736842

 **Ответ:**
 
 $\mu=65.3$
 
 $S=30.82$
 
 $S^{2}=950.11$
 
 $S_{unbiased.}^{2}=1000.12$

### 2) В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [7]:
p11 = combinations(5, 2) / combinations(8, 2)
p12 = combinations(5, 1) * combinations(7, 3) / combinations(12, 4)
p1 = p11 * p12
p1

0.12626262626262627

In [8]:
p21 = combinations(5, 1) * combinations(3, 1) / combinations(8, 2)
p22 = combinations(5, 2) * combinations(7, 2) / combinations(12, 4)
p2 = p21 * p22
p2

0.22727272727272727

In [9]:
p31 = combinations(3, 2) / combinations(8, 2)
p32 = combinations(5, 3) * combinations(7, 1) / combinations(12, 4)
p3 = p31 * p32
p3

0.01515151515151515

In [10]:
P = p1 + p2 + p3
np.round(P*100, decimals=2)

36.87

**Ответ: 36.87%**

### 3) На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Событием  𝐴  будем считать попадание в мишень, а событиями  ${B_1}$, ${B_2}$  и  ${B_3}$  — что выстрел совершил первый, второй или третий спортсмен. Вероятность выбора одного из трех спортсменов: $$P({B_1})=P({B_2})=P({B_3})=\frac{1}{3}$$

Вероятность попадания стрелком, при условии, что его выбрали:

$$P(A|B_{1})=0.9$$

$$P(A|B_{2})=0.8$$

$$P(A|B_{3})=0.6$$

Для решения воспользуемся формулой Байеса $$P(B|A)=\frac{P(B)\cdot P(A|B)}{P(A)}$$

Найдем полную вероятность $P(A)=P(B_{1})\cdot P(A|B_{1})+P(B_{2})\cdot P(A|B_{2})+ \dots +P(B_{n})\cdot P(A|B_{n})$

In [11]:
P_A = 1/3 * 0.9 + 1/3 * 0.8 + 1/3 * 0.6
P_A

0.7666666666666666

**a) выстрел произведен первым спортсменом $$P(B_{1}|A)=\frac{P(B_{1})\cdot P(A|B_{1})}{P(A)}$$**

In [12]:
P_B1_A = 1/3 * 0.9 / P_A
np.round(P_B1_A*100, decimals=2)

39.13

**б) вторым**

In [13]:
P_B2_A = 1/3 * 0.8 / P_A
np.round(P_B2_A*100, decimals=2)

34.78

**в) третьим**

In [14]:
P_B3_A = 1/3 * 0.6 / P_A
np.round(P_B3_A*100, decimals=2)

26.09

**Ответ: a) первым 39.13%; б) вторым 34.78%; в) третьим 26.09%**

### 4) В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Предположим, что кол-во студентов это $x$. По условию в каждом факультете количество студентов равно:

факультет A $=x$

факультет B $=x$

факультет С $=2x$

Общее кол-во студентов: $X_{total}= x + x + 2x = 4x$

Вероятность случайным образом выбрать студента из 3-x факультетов равна: 

факультет А $P(B_{1})=\frac{x}{4x}=\frac{1}{4}$

факультет В $P(B_{2})=\frac{x}{4x}=\frac{1}{4}$

факультет С $P(B_{3})=\frac{2x}{4x}=\frac{1}{2}$

Вероятности сдачи сессии студентами по всем факультетам:

студент из факультета А $P(A|B_{1})=0.8$

студент из факультета В $P(A|B_{2})=0.7$

студент из факультета С $P(A|B_{3})=0.9$

Для решения воспользуемся формулой Байеса $$P(B|A)=\frac{P(B)\cdot P(A|B)}{P(A)}$$

Найдем полную вероятность $P(A)=P(B_{1})\cdot P(A|B_{1})+P(B_{2})\cdot P(A|B_{2})+ \dots +P(B_{n})\cdot P(A|B_{n})$

In [15]:
P_A = 1/4 * 0.8 + 1/4 * 0.7 + 1/2 * 0.9
P_A

0.825

**а) Студент сдал первую сессию. Вероятность, что он учится на факультете A $P(B_{1}|A)$:**

In [16]:
P_B1_A = 1/4 * 0.8 / P_A
np.round(P_B1_A*100, decimals=2)

24.24

**б) на факультете В $P(B_{2}|A)$**

In [17]:
P_B2_A = 1/4 * 0.7 / P_A
np.round(P_B2_A*100, decimals=2)

21.21

**в) на факультете С $P(B_{3}|A)$**

In [18]:
P_B3_A = 1/2 * 0.9 / P_A
np.round(P_B3_A*100, decimals=2)

54.55

**Ответ: a) на факультете A 24.24%; б) на факультете В 21.21%; в) на факультете C 54.55%**

### 5) Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

Даны 3 детали - A, B, C.

Вероятности выйти из строя в первый месяц:

$P(A)=0.1$

$P(B)=0.2$

$P(C)=0.25$

In [19]:
P_A = 0.1
P_B = 0.2
P_C = 0.25

Найдем вероятности выйти из строя в первый месяц:

**а) всех деталей**

In [20]:
P1 = P_A * P_B * P_C
np.round(P1*100, decimals=2)

0.5

**б) только двух деталей**

In [21]:
P2 = (P_A * P_B * (1-P_C)) + ((1-P_A) * P_B * P_C) + (P_A * (1-P_B) * P_C)
np.round(P2*100, decimals=2)

8.0

**в) хотя бы одной детали**

От противного:

In [22]:
P3 = 1 - (0.9) * (0.8) * (0.75)
np.round(P3*100, decimals=2)

46.0

Или так:

In [23]:
p1 = (P_A * (1-P_B) * (1-P_C)) + ((1-P_A) * P_B * (1-P_C)) + ((1-P_A) * (1-P_B) * P_C)
p1

0.375

In [24]:
p2 = (P_A * P_B * (1-P_C)) + ((1-P_A) * P_B * P_C) + (P_A * (1-P_B) * P_C)
p2

0.08000000000000002

In [25]:
p3 = P_A * P_B * P_C
p3

0.005000000000000001

In [26]:
np.round((p1+p2+p3)*100, decimals=2)

46.0

**г) от одной до двух деталей**

In [27]:
P4 = p1 + p2
P4*100

45.5

**Ответ:**

а) все детали 0.5%

б) только две детали 8%

в) хотя бы одна деталь 46%

г) от одной до двух деталей 45.5%